apps/src/templates/ReCaptchaDialog.jsx

```diff
@@ -39,15 +39,22 @@ export default class ReCaptchaDialog extends React.Component {
     this.onCaptchaExpiration = this.onCaptchaExpiration.bind(this);
   }
 
-  componentDidMount() {
-    //Add reCaptcha and associated callbacks.
-    const script = document.createElement('script');
-    script.src = 'https://www.google.com/recaptcha/api.js';
-    script.id = 'captcha';
-    window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
-    window.onCaptchaExpired = () => this.onCaptchaExpiration();
-    script.onload = () => this.setState({loadedCaptcha: true});
-    document.body.appendChild(script);
+  componentDidUpdate(prevProps) {
+    // If the dialog goes from unopen to open, load the script and disabled submit button
+    if (this.props.isOpen && !prevProps.isOpen) {
+      this.setState({submitButtonEnabled: false});
+      //Add reCaptcha script and associated callbacks.
+      const script = document.createElement('script');
+      script.src = 'https://www.google.com/recaptcha/api.js';
+      script.id = 'captcha';
+      script.async = true;
+      window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
+      window.onCaptchaExpired = () => this.onCaptchaExpiration();
+      document.head.appendChild(script);
+      script.onload = () => {
+        this.setState({loadedCaptcha: true});
+      };
```

Comment:
Do any of these steps need to be cleaned up somewhere?

Reply:
See `onCloseCleanup` -- new function created to clean up the DOM and delete the callback functions so we don't end up with duplicate keys on `window`.